src/cache_cf.cc

```diff
@@ -1551,6 +1551,8 @@ parse_address(Ip::Address *addr)
         addr->setAnyAddr();
     else if ( (!strcmp(token,"no_addr")) || (!strcmp(token,"full_mask")) )
         addr->setNoAddr();
+    else if (!strcmp(token,"transparent"))
```

Comment:
Please do not ignore "transparent" label in unsupported contexts:

```
src/cf_parser.cci:        parse_address(&Config.Addrs.client_netmask);
src/cf_parser.cci:        parse_address(&Config.Wccp.router);
src/cf_parser.cci:        parse_address(&Config.Wccp.address);
src/cf_parser.cci:        parse_address(&Config.Wccp2.address);
src/cf_parser.cci:        parse_address(&Config.Addrs.snmp_incoming);
src/cf_parser.cci:        parse_address(&Config.Addrs.snmp_outgoing);
src/cf_parser.cci:        parse_address(&Config.Addrs.udp_incoming);
src/cf_parser.cci:        parse_address(&Config.Addrs.udp_outgoing);
src/cf_parser.cci:        parse_address(&Config.mcast_miss.addr);
```

I did not check all the details, but I would start by splitting parse_address() into two functions:

* parse_address(): Extracts the next token from the parser and calls parseAddressToken() to parse it
* parseAddressToken(): Parses the given token.

PR's parse_acl_address() will then not use parse_address(). Instead, it will extract the token, handle "transparent", and then call parseAddressToken().

Other/old parse_address() callers will remain unchanged.

The above plan also avoids adding a new PeekAtToken() call. We are trying to get rid of that bad hack...

Reply:
I have implemented this suggestion.